### PR TITLE
Hotfix 6.2.4 - Outbox Bugs

### PR DESCRIPTION
**Pull request by DavidBoike**
Connects to #117

Add EndpointName to stored MessageId, use straightforward delete for cleanup

### CODE REVIEW

**Comment by DavidBoike** [comment]
### Acceptance Test Results: 

#### [MsmqTransport](http://builds.particular.net/viewLog.html?buildId=129581&tab=buildResultsDiv&buildTypeId=NServiceBusNHibernate_2AcceptanceTests)

Both on Oracle, Both in NServiceBus.AcceptanceTests.Exceptions, both stack trace assertions??? These are the ATs that @SimonCropp took out right? 

* When_handler_and_Uow_End_throws.Should_receive_AggregateException_with_both_exceptions
* When_handler_throws.Should_receive_exception_from_handler

#### [SqlTransport](http://builds.particular.net/viewLog.html?buildId=129579&tab=buildResultsDiv&buildTypeId=NServiceBusNHibernate_AcceptanceTestsSqlServer)

Both in MSSQL Tests:

* NServiceBus.AcceptanceTests.Sagas.When_reply_from_a_finder.Should_be_received_by_handler
  * Need to test locally
* NServiceBus.AcceptanceTests.ScaleOut.When_individualization_is_enabled.Should_use_the_configured_differentiator
  * This is the one that seems totally related to SQL Transport and completely unrelated from Outbox/NHibernate/persistence in any way

**Comment by SimonCropp** [comment]
@DavidBoike yeah just kill those stack trace tests. The whole dir

**Comment by DavidBoike** [comment]
MsmqTransport is now green.

[SqlTransport](http://builds.particular.net/viewLog.html?buildId=130251&tab=buildResultsDiv&buildTypeId=NServiceBusNHibernate_AcceptanceTestsSqlServer) is a different story. There are a bunch of failed Oracle tests that failed this time, but succeeded before with no changes. Can the Oracle tests be ignored - why would you use Oracle persistence with SqlServer transport?

That leaves the same two failing tests for SqlTransport as noted above.

**Comment by DavidBoike** [comment]
The first failing test on SqlTransport is a test that, when run in isolation with a clean database, will succeed. I am going to ignore it.

That leaves the discriminator test. Can I get some help with that? @johnsimons @SzymonPobiega @SimonCropp @andreasohlund (Top contributors to SqlServer transport)

**Comment by DavidBoike** [comment]
@SzymonPobiega says go ahead and ignore all the tests currently failing. So this should be good to go for a release. @Particular/engineers please review.

File: src/NServiceBus.NHibernate/Outbox/OutboxPersister.cs [hunk 1]
```diff
@@ -75,10 +83,11 @@ public void SetAsDispatched(string messageId)
                 {
                     using (var tx = session.BeginTransaction(IsolationLevel.ReadCommitted))
                     {
-                        var queryString = string.Format("update {0} set Dispatched = true, DispatchedAt = :date where MessageId = :messageid And Dispatched = false",
+                        var queryString = string.Format("update {0} set Dispatched = true, DispatchedAt = :date where MessageId IN ( :messageid, :qualifiedMsgId ) And Dispatched = false",
                             typeof(OutboxRecord));
                         session.CreateQuery(queryString)
                             .SetString("messageid", messageId)
+                            .SetString("qualifiedMsgId", EndpointQualifiedMessageId(messageId))
```
Comment (SimonCropp) [inline src/NServiceBus.NHibernate/Outbox/OutboxPersister.cs]: `messageid` and `...MsgId`. the OCD it burns

**Comment by SzymonPobiega** [comment]
To be precise, the failing tests we are talking about are
 * A scale-out tests added in the core acceptance test suite but not currently supported by the SQL Server transport. In SQL Server repo the test has been removed but here we can't do that
 * A SQL Server transport - Oracle persistence tests that has been flaky since the down of time

**Comment by SzymonPobiega** [comment]
:+1: for merging.

**Pull request merged**